platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/Hash.java

```diff
@@ -221,26 +193,15 @@ public boolean equals(final Object obj) {
             return false;
         }
 
-        return digestType.id() == that.digestType.id() && Arrays.equals(value, that.value);
-    }
-
-    /**
-     * Check if the bytes of this hash are equal to the bytes supplied.
-     *
-     * @param bytes
-     * 		the bytes to compare
-     * @return true if the bytes are equal, false otherwise
-     */
-    public boolean equalBytes(@Nullable final Bytes bytes) {
-        return bytes != null && value.length == bytes.length() && bytes.contains(0, value);
+        return digestType.id() == that.digestType.id() && Objects.equals(bytes, that.bytes);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public int hashCode() {
-        return ((value[0] << 24) + (value[1] << 16) + (value[2] << 8) + (value[3]));
+        return bytes.getInt(0);
     }
```

Comment:
Minor: I understand that the new implementation implicitly assumes that PBJ Bytes are big endian by default, and so it is equivalent to the old one. Two points:
1. It might make the code less cryptic if we avoid the implicit assumption and use the `getInt(offset, byteOrder)` version of the method explicitly. Or, if we want the maximum performance, perhaps a comment here in code suggesting that we're aware of this assumption might be helpful.
2. Out of curiosity: why does this hashCode() implementation ignore the rest of the bytes in the bytes array? Is this done for performance reasons only? Does it make sense to compute the hash using all the bytes instead? Note that Bytes.hashCode() does this already. Although I'm unsure if anything would break if this implementation changes - do we persist this hash value anywhere?

Reply:
+1

Reply:
This comment made me find [an issue](https://github.com/hashgraph/pbj/issues/256) in the `Bytes` class.

1. I don't think we care if its big endian or little, so long as its consistent. And not reversing the bytes is better for performance, which is important. I will add a comment.
2. Performance is very important for a few different use cases of this class. For gossip, we make huge hash sets of events, and when profiling a long time ago, this made a huge difference. Apart from performance, we are dealing with a cryptographic hash, so the first four bytes are just as optimal as anything else we can compute. I don't see why changing the implementation would affect anything, I think the only important thing is that this method is consistent at runtime.

I will add comments to the code with explanations for everything.

Reply:
Thanks for filing the issue, but there's no bugs - I've commented in that ticket and provided more details.

Also, thanks for the additional context and the comment in code. It makes sense to me.